services/api/service.go

```diff
@@ -484,94 +486,134 @@ func (api *RelayAPI) handleRegisterValidator(w http.ResponseWriter, req *http.Re
 		"mevBoostV": common.GetMevBoostVersionFromUserAgent(ua),
 	})
 
-	respondError := func(code int, msg string) {
-		log.Warn("bad request: ", msg)
-		api.RespondError(w, code, msg)
-	}
-
 	start := time.Now()
 	registrationTimeUpperBound := start.Add(10 * time.Second)
 
-	registrations := []types.SignedValidatorRegistration{}
+	numRegTotal := 0
+	numRegProcessed := 0
+	numRegActive := 0
 	numRegNew := 0
+	processingStoppedByError := false
 
-	if err := json.NewDecoder(req.Body).Decode(&registrations); err != nil {
-		respondError(http.StatusBadRequest, "failed to decode payload")
+	respondError := func(code int, msg string) {
```

Comment:
y not a pure function accepting response writer?

Reply:
fixed after our call to make it more functional